nwbinspector/nwbinspector.py

```diff
@@ -172,22 +197,21 @@ def inspect_nwb(
                     unorganized_results["PYNWB_VALIDATION"].append(message)
             nwbfile = io.read()
             check_results = list()
-            for importance, checks_per_object_type in checks.items():
+            for importance, check_functions in checks.items():
                 if importance.value >= importance_threshold.value:
-                    for check_object_type, check_functions in checks_per_object_type.items():
-                        for nwbfile_object in nwbfile.objects.values():
-                            if issubclass(type(nwbfile_object), check_object_type):
-                                for check_function in check_functions:
-                                    if ignore is not None and check_function.__name__ in ignore:
-                                        continue
-                                    if select is not None and check_function.__name__ not in select:
-                                        continue
-                                    output = check_function(nwbfile_object)
-                                    if output is not None:
-                                        if isinstance(output, Iterable):
-                                            check_results.extend(output)
-                                        else:
-                                            check_results.append(output)
+                    for nwbfile_object in nwbfile.objects.values():
+                        for check_function in check_functions:
+                            if issubclass(type(nwbfile_object), check_function.neurodata_type):
+                                if ignore is not None and check_function.__name__ in ignore:
+                                    continue
+                                if select is not None and check_function.__name__ not in select:
+                                    continue
```

Comment:
```suggestion
                    for nwbfile_object in nwbfile.objects.values():
                        for check_function in check_functions:
                            if ignore is not None and check_function.__name__ in ignore:
                                 continue
                            if select is not None and check_function.__name__ not in select:
                                 continue
                            if issubclass(type(nwbfile_object), check_function.neurodata_type):
```

Minor suggestion for more speed; can ignore `if issubclass` call entirely if this check name isn't selected or is ignored.